src/dali/plugin/input/rest/kucoin_futures.py

```diff
@@ -0,0 +1,280 @@
+
+                    # To do: need to replace transaction type from INCOME to GAIN
+                    # GAIN currently not implemented in rp2
+                    in_transactions.append(
+                        InTransaction(
+                            plugin=self.plugin_name(),
+                            unique_id=unique_id,
+                            raw_data=json.dumps(item),
+                            timestamp=timestamp,
+                            asset=currency,
+                            exchange=self.exchange_name(),
+                            holder=self.account_holder,
+                            transaction_type=Keyword.INCOME.value,
+                            spot_price=Keyword.UNKNOWN.value,
+                            crypto_in=str(amount_value),
+                            crypto_fee=None,
+                            fiat_in_no_fee=None,
+                            fiat_in_with_fee=None,
+                            fiat_fee=None,
+                            notes="Futures gain"
+                        )
+                    )
+
+                else:
+                    # Loss
+                    # Example json
+                    # {
+                    #   "time": 1644444400000,
+                    #   "type": "RealisedPNL",
+                    #   "amount": -66.66666666,
+                    #   "fee": 0.0,
+                    #   "accountEquity": 12345.67890123,
+                    #   "status": "Completed",
+                    #   "remark": "ETHUSDTM",
+                    #   "offset": 123456,
+                    #   "currency": "USDT"
+                    # }
+
+                    # To do: need to replace transaction type from SELL to LOSS
+                    # LOSS currently not implemented in rp2
+                    out_transactions.append(
+                        OutTransaction(
+                            plugin=self.plugin_name(),
+                            unique_id=unique_id,
+                            raw_data=json.dumps(item),
+                            timestamp=timestamp,
+                            asset=currency,
+                            exchange=self.exchange_name(),
+                            holder=self.account_holder,
+                            transaction_type=Keyword.SELL.value,
+                            spot_price="0",
+                            crypto_out_no_fee=str(-amount_value),
+                            crypto_fee="0",
+                            crypto_out_with_fee=str(-amount_value),
+                            fiat_out_no_fee=None,
+                            fiat_fee=None,
+                            notes="Futures loss",
+                        )
+                    )
+
+    def _process_implicit_api(
+        self,
+        in_transactions: List[InTransaction],
+        out_transactions: List[OutTransaction],
+        intra_transactions: List[IntraTransaction],
+    ) -> None:
+        end_at: int = int(time.time() * 1000)
```

Comment:
Can you add a new protected variable and change this line to make it more clear.
Above:
```
_MS_IN_SECOND: int = 1000
```
Here:
```
end_at: int = int(time.time() * _MS_IN_SECOND)
```

Reply:
fixed